lib/BloqadeQMC/src/ising/Rydberg.jl

```diff
@@ -1,23 +1,23 @@
 using Base.Iterators
 
-
 abstract type AbstractRydberg{O <: AbstractOperatorSampler} <: AbstractLTFIM{O} end
 
-struct Rydberg{O,M <: UpperTriangular{Float64},UΩ <: AbstractVector{Float64}, Uδ <: AbstractVector{Float64}, L <: Lattice} <: AbstractRydberg{O}
+struct Rydberg{O,M <: AbstractMatrix{Float64},UΩ <: AbstractVector{Float64}, Uδ <: AbstractVector{Float64}, A} <: AbstractRydberg{O}
     op_sampler::O
     V::M
     Ω::UΩ
     δ::Uδ
-    lattice::L
+    atoms::A
     energy_shift::Float64
 end
-nspins(H::Rydberg) = nspins(H.lattice)
+
+nspins(H::Rydberg) = length(H.atoms)
 
 @inline diagonaloperator(::Type{<:AbstractRydberg}) = Diagonal([0, 1])
 @inline diagonaloperator(H::AbstractRydberg) = diagonaloperator(typeof(H))
 
 
-function make_prob_vector(H::Type{<:AbstractRydberg}, V::UpperTriangular{T}, Ω::AbstractVector{T}, δ::AbstractVector{T}; epsilon=0.0) where T
+function make_prob_vector(H::Type{<:AbstractRydberg}, V::AbstractMatrix{T}, Ω::AbstractVector{T}, δ::AbstractVector{T}; epsilon=0.0) where T
```

Comment:
why are we changing it from `UpperTriangular` to a more general case?

Reply:
I would need to go in and modify the `rydberg_interaction_matrix` to make this work since that returns a full Matrix.